### PR TITLE
[CELEBORN-1477][CIP-9][FOLLOWUP] User guide for `/api/v1` migration

### PR DESCRIPTION
**Pull request by turboFei**
### What changes were proposed in this pull request?

Provide the user migration guide for /api/v1.

### Why are the changes needed?

For migration.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Verified the api mapping in swagger.

### CODE REVIEW

**Comment by turboFei** [comment]
cc @pan3793 @SteNicholas @RexXiong

File: docs/migration.md [hunk 1]
```diff
@@ -21,6 +21,52 @@ license: |
 
 # Migration Guide
 
+# Upgrading from 0.5 to 0.6
+- 
+- Since 0.6.0, Celeborn has introduced a new API namespace: /api/v1, which uses the application/json media type for requests and responses.
+   The `celeborn-opanapi-client` SDK is also available to help users interact with the new APIs.
```
Comment (cxzl25) [inline docs/migration.md:27]: celeborn-openapi-client

Comment (turboFei) [reply docs/migration.md]: thanks, fixed

**Comment by pan3793** [comment]
API => RESTful API

**Comment by turboFei** [comment]
> API => RESTful API

Done.

**Review by pan3793 — APPROVED** [review]

**Review by cxzl25 — APPROVED** [review]

File: docs/migration.md [hunk 2]
```diff
@@ -21,6 +21,53 @@ license: |
...
+    | GET /masterGroupInfo     | GET /api/v1/masters                      |                                                  |
+    | GET /workerInfo          | GET /api/v1/workers                      |                                                  |
+    | GET /lostWorkers         | GET /api/v1/workers                      | get the lostWorkers field in response            |
+    | GET /excludedWorkers     | GET /api/v1/workers                      | get the excludedWorkers field in response        |
+    | GET /shutdownWorkers     | GET /api/v1/workers                      | get the shutdownWorkers filed in response        |
+    | GET /decommissionWorkers | GET /api/v1/workers                      | get the decommissioningWorkers filed in response |
+    | POST /exclude            | POST /api/v1/workers/exclude             |                                                  |
+    | GET /workerEventInfo     | GET /api/v1/workers/events               |                                                  |
+    | POST /sendWorkerEvent    | POST /api/v1/workers/events              |                                                  |
```
Comment (SteNicholas) [inline docs/migration.md:50]:
```suggestion
    | POST /sendWorkerEvent    | POST /api/v1/workers/events              |  the `type` parameter changed to  `eventType`                                               |
```

Comment (turboFei) [reply docs/migration.md:50]: thanks, I think it is better to check the swagger UI for details, here we only show the api mapping.

Comment (turboFei) [reply docs/migration.md:50]: image

Comment (SteNicholas) [reply docs/migration.md:50]: The users only check the details in document, not in idea. Right?

Comment (turboFei) [reply docs/migration.md:50]: Use can access the full rest api doc: https://github.com/apache/celeborn/blob/main/docs/webapi.md

For swagger UI:
1. they can check the swagger http://<http-host>:<http-port>/swagger 
2. They can locally preview the OpenAPI specification YAML file in IDE, or use the [Swagger Editor](https://editor.swagger.io/) online for visualization.

Comment (turboFei) [reply docs/migration.md:50]: And here both the request and response are changed, without swagger, it is difficult to describe the API change.

For example: for `/exclude` api:

Before:
```
curl -X POST http://host:port/exclude  -H "Content-Type: application/x-www-form-urlencoded" \
  -d "add=localhost:9200:9201:9202:9203"
```

After:

```
curl -X 'POST' \
  'http://localhost:9098/api/v1/workers/exclude' \
  -H 'accept: application/json' \
  -H 'Content-Type: application/json' \
  -d '{
  "add": [
    {
      "host": "localhost",
      "rpcPort": 9201,
      "pushPort": 9202,
      "fetchPort": 9203,
      "replicatePort": 9204
    }
  ]
}'
```

Comment (turboFei) [reply docs/migration.md:50]: If needed, I can try to generate the docs with openapi-generator, so that we can maintain the RESTapi easily.

**Review by SteNicholas — APPROVED** [review]

**Comment by SteNicholas** [comment]
Merged to main(v0.6.0).

**Comment by turboFei** [comment]
thanks @SteNicholas @pan3793 @cxzl25

**Pull request closed without merging**